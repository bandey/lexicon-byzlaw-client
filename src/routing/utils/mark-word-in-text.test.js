import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

import enzymeToJson from 'enzyme-to-json';

import markWordInText from './mark-word-in-text.js';

it('mark word in first position', () => {
  const rendered = mount(
    <React.Fragment>
      {markWordInText('first', ' first last ')}
    </React.Fragment>
  );

  // console.log(rendered.debug());

  expect(enzymeToJson(rendered)).toMatchSnapshot();
});

it('mark word in middle position', () => {
  const rendered = mount(
    <React.Fragment>
      {markWordInText('middle', ' first middle last ')}
    </React.Fragment>
  );

  // console.log(rendered.debug());

  expect(enzymeToJson(rendered)).toMatchSnapshot();
});

it('mark word in last position', () => {
  const rendered = mount(
    <React.Fragment>
      {markWordInText('last', ' first last ')}
    </React.Fragment>
  );

  // console.log(rendered.debug());

  expect(enzymeToJson(rendered)).toMatchSnapshot();
});

it('no marking, if word is not found', () => {
  const rendered = mount(
    <React.Fragment>
      {markWordInText('middle', ' first last ')}
    </React.Fragment>
  );

  // console.log(rendered.debug());

  expect(enzymeToJson(rendered)).toMatchSnapshot();
});

it('mark several words', () => {
  const rendered = mount(
    <React.Fragment>
      {markWordInText('middle', ' first middle middle last ')}
    </React.Fragment>
  );

  // console.log(rendered.debug());

  expect(enzymeToJson(rendered)).toMatchSnapshot();
});
