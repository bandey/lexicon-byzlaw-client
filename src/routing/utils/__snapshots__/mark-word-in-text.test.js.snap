// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mark several words 1`] = `
Array [
  <styled.span
    key="1"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       first 
    </span>
  </styled.span>,
  <styled.span
    key="2"
    marked={true}
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm hZjGXT"
    >
      middle
    </span>
  </styled.span>,
  <styled.span
    key="3"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       
    </span>
  </styled.span>,
  <styled.span
    key="4"
    marked={true}
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm hZjGXT"
    >
      middle
    </span>
  </styled.span>,
  <styled.span
    key="5"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       last 
    </span>
  </styled.span>,
]
`;

exports[`mark word in first position 1`] = `
Array [
  <styled.span
    key="1"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       
    </span>
  </styled.span>,
  <styled.span
    key="2"
    marked={true}
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm hZjGXT"
    >
      first
    </span>
  </styled.span>,
  <styled.span
    key="3"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       last 
    </span>
  </styled.span>,
]
`;

exports[`mark word in last position 1`] = `
Array [
  <styled.span
    key="1"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       first 
    </span>
  </styled.span>,
  <styled.span
    key="2"
    marked={true}
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm hZjGXT"
    >
      last
    </span>
  </styled.span>,
  <styled.span
    key="3"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       
    </span>
  </styled.span>,
]
`;

exports[`mark word in middle position 1`] = `
Array [
  <styled.span
    key="1"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       first 
    </span>
  </styled.span>,
  <styled.span
    key="2"
    marked={true}
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm hZjGXT"
    >
      middle
    </span>
  </styled.span>,
  <styled.span
    key="3"
    whiteSpace={true}
  >
    <span
      className="sc-AxjAm xrrgJ"
    >
       last 
    </span>
  </styled.span>,
]
`;

exports[`no marking, if word is not found 1`] = `
<styled.span
  key="1"
  whiteSpace={true}
>
  <span
    className="sc-AxjAm xrrgJ"
  >
     first last 
  </span>
</styled.span>
`;
